client/js/geometry.js

```diff
@@ -18,12 +18,58 @@ export function getOffset(origin, target) {
   return {x: target.x - origin.x, y: target.y - origin.y}
 }
 
-export function applyTransformedOffset(origin, offset, scale, rotation) {
-  if(rotation % 360 == 0) {
-    return {x: origin.x + offset.x * scale, y: origin.y + offset.y * scale}
-  } else {
-    const dist = distance({x:0,y:0},offset) * scale;
-    const angle = Math.atan2(offset.y ,offset.x) + rotation * Math.PI / 180;
-    return {x: origin.x + dist * Math.cos(angle), y: origin.y + dist * Math.sin(angle)};
+export function getScreenTransform(elem) {
+  let transform = new DOMMatrix();
+  let t = new DOMPoint(10, 10);
+  while (elem) {
+    transform.preMultiplySelf(getElementTransform(elem));
+    elem = elem.offsetParent;
   }
+  return transform;
+}
+
+function parseLengths(str) {
+  return str.split(' ').map(s => parseFloat(s))
```

Comment:
I decided to simply split on ' ' here (compared to the initial commit) since the computed `transformOrigin` and `perspectiveOrigin` can always be expected to be `###px ###px` and it should be more efficient than a regular expression.